extensions/ssh-tunnel-manager/src/screens/ListScreen.tsx

```diff
@@ -0,0 +1,203 @@
+  const [listData, setListData] = useState<ListData[]>(cachedList ?? []);
+  const [selectedId, setSelectedId] = useState<string | null>(null);
+
+  const { setTunnelParams, isLoading } = useTunelCmd(
+    listData,
+    setListData,
+    alivePidSet,
+    defaultTunnelParams,
+    shouldEstablish
+  );
+
+  const { killTunnel, setPidToKill } = useKillTunnelCmd();
+
+  const { refreshList } = useSshPidList();
+
+  const tunnelTrigger = () => {
+    const tunnelItem = listData.find((i) => i.name === selectedId);
+    if (!tunnelItem?.pid) {
+      setTunnelParams((params) => {
+        const res = { ...params, ...tunnelItem } as Partial<ListData>;
+        delete res.pid;
+        return res as Values;
+      });
+      showToast({ title: "Tunnel established" });
+    } else {
+      setPidToKill(tunnelItem.pid);
+      alivePidSet.delete(tunnelItem.pid);
+      cache.set(Storage.AlivePidList, JSON.stringify(Array.from(alivePidSet)));
+      tunnelItem.pid = null;
+      setListData([...listData]);
+      showToast({ title: "Tunnel Closed" });
+    }
+  };
+
+  const deleteTunnel = () => {
+    confirmAlert({
+      title: "Confirm",
+      message: "Tunnel will be deleted, are you sure?",
+      primaryAction: {
+        title: "Delete",
+        onAction: () => {
+          const itemToDelete = listData.find((i) => i.name === selectedId);
+          const newListData = listData.filter((i) => i.name !== selectedId);
+          setListData(newListData);
+          if (itemToDelete?.pid) {
+            setPidToKill(itemToDelete.pid);
+            alivePidSet.delete(itemToDelete.pid);
+            cache.set(Storage.AlivePidList, JSON.stringify(Array.from(alivePidSet)));
+            killTunnel();
+          }
+          markUsed();
+          setTunnelParams({});
+          showToast({ title: "Tunnel deleted" });
+        },
+      },
+    });
+  };
+
+  const generateListData = () => {
+    listData.forEach((data) => {
+      if (data.pid && !alivePidSet.has(data.pid)) data.pid = null;
+    });
+    if (shouldEstablish.current === false && !listData.find((i) => i.name === defaultTunnelParams.name)) {
+      listData.push({ ...(defaultTunnelParams as Values), pid: null });
+    } else if (shouldEstablish.current === true && !listData.find((i) => i.name === defaultTunnelParams.name)) {
+      if (defaultTunnelParams.name) {
+        listData.push({ ...(defaultTunnelParams as Values), pid: null });
+        setTunnelParams({ ...defaultTunnelParams });
+      }
+    }
+    cache.set(Storage.List, JSON.stringify(listData));
+  };
+
+  generateListData();
+
+  return (
+    <List
+      isLoading={isLoading}
+      isShowingDetail={listData.length !== 0}
+      searchBarPlaceholder={"Search Tunnels"}
+      onSelectionChange={(id) => {
+        setSelectedId(id);
+      }}
+      actions={
+        <ActionPanel>
+          <Action title="Refresh List" onAction={refreshList} />
+          <Action title="Create Tunnel" onAction={toCreate} />
```

Comment:
Double of same action?

Reply:
emm, i want to use same shortcut whenever list is empty or not, do you have any suggestions?

Reply:
I was talking about this

![Raycast-EHBJvKjs@2x](https://user-images.githubusercontent.com/38033107/213437471-351b3786-783b-4381-921e-183ddea525be.png)

<img width="423" alt="Raycast-axxc0DqP@2x" src="https://user-images.githubusercontent.com/38033107/213437509-dcbaca52-1e78-4e2d-8b7b-8b53d32025dd.png">

Reply:
same for refresh 😊

Reply:
I know, but i don't know how to override the default shortcut enter and ⌘+enter

Reply:
The bind should work but it won't show it for the first two items - that's how it works - the first two items will show enter and ⌘+enter

Reply:
ok, deleted